bmp-string.go

```diff
@@ -12,8 +13,12 @@ func bmpString(s string) ([]byte, error) {
 	//	  EncodeRune returns 0xfffd if the rune does not need special encoding
 	//  - the above RFC provides the info that BMPStrings are NULL terminated.
 
-	rv := make([]byte, 0, 2*len(s)+2)
-	for _, c := range []rune(s) {
+	rv := make([]byte, 0, 2*len(utf8String)+2)
```

Comment:
`utf8.RuneCount` would be more accurate than `len(utf8String)`

(on the other hand, it calls DecodeRun, so likely slower)